pkg/alertmanager/statefulset.go

```diff
@@ -666,6 +666,19 @@ func makeStatefulSetSpec(logger log.Logger, a *monitoringv1.Alertmanager, config
 		alertmanagerURIScheme = "https"
 	}
 
+	securityContext := &v1.SecurityContext{
+		ReadOnlyRootFilesystem:   ptr.To(true),
+		AllowPrivilegeEscalation: ptr.To(false),
+		RunAsNonRoot:             ptr.To(true),
+		RunAsUser:                ptr.To(int64(1000)),
```

Comment:
This would be a breaking change for some users.

Reply:
@simonpasquier previously i have checked  all e2e tests for cofig-reloader, they are working fine. then i have updated the stuffs for other containers

Reply:
```
 		
ReadOnlyRootFilesystem:   ptr.To(true),
		AllowPrivilegeEscalation: ptr.To(false),
		RunAsNonRoot:             ptr.To(true),
		RunAsUser:                ptr.To(int64(1000)),
```
@simonpasquier can you explain why this will result in break

Reply:
iirc the images of prometheus, thanios-ruler, and alertmanager allows runasNonRoot and they didnt have userId specified

Reply:
people might use custom images (e.g. not from upstream).

Reply:
@simonpasquier, it is necessary to specify userId, for the feild RunAsNonRoot=true, how we can get the userID in the user uses custom images

Reply:
or we can ask user to specify userID while the user is specifying custom image
```
	// +optional
	Image *string `json:"image,omitempty"`
        UserID *string
 ```